pkg/storageincentives/redistribution/inclusionproof.go

```diff
@@ -0,0 +1,122 @@
+// Copyright 2023 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Used for inclusion proof utilities
+
+package redistribution
+
+import (
+	"encoding/binary"
+	"encoding/hex"
+
+	"github.com/ethersphere/bee/pkg/bmt"
+	"github.com/ethersphere/bee/pkg/soc"
+	"github.com/ethersphere/bee/pkg/storer"
+	"github.com/ethersphere/bee/pkg/swarm"
+)
+
+type ChunkInclusionProofs struct {
+	A ChunkInclusionProof `json:"proof1"`
+	B ChunkInclusionProof `json:"proof2"`
+	C ChunkInclusionProof `json:"proofLast"`
+}
+
+// ChunkInclusionProof structure must exactly match
+// corresponding structure (of the same name) in Redistribution.sol smart contract.
+// github.com/ethersphere/storage-incentives/blob/ph_f2/src/Redistribution.sol
+// github.com/ethersphere/storage-incentives/blob/master/src/Redistribution.sol (when merged to master)
+type ChunkInclusionProof struct {
+	ProofSegments  []string     `json:"proofSegments"`
+	ProveSegment   string       `json:"proveSegment"`
+	ProofSegments2 []string     `json:"proofSegments2"`
+	ProveSegment2  string       `json:"proveSegment2"`
+	ChunkSpan      uint64       `json:"chunkSpan"`
+	ProofSegments3 []string     `json:"proofSegments3"`
+	PostageProof   PostageProof `json:"postageProof"`
+	SocProof       []SOCProof   `json:"socProof"`
+}
+
+// SOCProof structure must exactly match
+// corresponding structure (of the same name) in Redistribution.sol smart contract.
+type PostageProof struct {
+	Signature string `json:"signature"`
+	PostageId string `json:"postageId"`
+	Index     string `json:"index"`
+	TimeStamp string `json:"timeStamp"`
+}
+
+// SOCProof structure must exactly match
+// corresponding structure (of the same name) in Redistribution.sol smart contract.
+type SOCProof struct {
+	Signer     string `json:"signer"`
+	Signature  string `json:"signature"`
+	Identifier string `json:"identifier"`
+	ChunkAddr  string `json:"chunkAddr"`
+}
+
+// Transforms arguments to ChunkInclusionProof object
+func NewChunkInclusionProof(proofp1, proofp2 bmt.Proof, proofp3 bmt.Proof, sampleItem storer.SampleItem) (ChunkInclusionProof, error) {
```

Comment:
This `proofp1, proofp2 bmt.Proof, proofp3 bmt.Proof` can be rewriten as `proofp1, proofp2, proofp3 bmt.Proof`.